packages/components/src/Filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import { Dropdown, DropdownToggle, Level, DropdownProps } from '@patternfly/react-core';
+import FilterInput from './FilterInput.js';
+import './filter-dropdown.scss';
+
+export interface FilterDropdownProps extends Omit<DropdownProps, 'onSelect'> {
+  addFilter: (filterName: string | number, selectedValue: string | number, type: string) => void;
+  removeFilter: (filterName: string | number, selectedValue: string | number) => void;
+  hideCategories: [];
+  filters: object;
+  filterCategories: [
+    {
+      title: string;
+      type: string;
+      urlParam: string;
+      values: [
+        {
+          label: React.ReactNode;
+          value: string | number;
+        }
+      ];
+    }
+  ];
+  onSelect: ((event?: React.SyntheticEvent<HTMLDivElement, Event> | undefined) => void) | undefined;
+  label: string;
+  ouiaId: string;
+  ouiaSafe: boolean;
+}
+
+const FilterDropdown: React.FunctionComponent<FilterDropdownProps> = ({
+  hideCategories,
+  filters,
+  filterCategories,
+  onSelect,
+  label,
+  ouiaId,
+  ouiaSafe,
+  addFilter,
+  removeFilter,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const ouiaStateId = 'RHI/FilterDropdown';
+
+  const addRemoveFilters = (selectedValue: string | number, filterName: string | number, type: string, isChecked: boolean) => {
+    switch (type) {
+      case 'checkbox':
+        isChecked ? addFilter(filterName, selectedValue, type) : removeFilter(filterName, selectedValue);
+        break;
+      case 'radio':
+        addFilter(filterName, selectedValue, type);
+        break;
+    }
+  };
+
+  const onToggle = (isOpen: boolean) => {
+    setIsOpen(!isOpen);
```

Comment:
I think the isOpen value should not be negated.

Reply:
Is this even needed, can I just use onToggle and isOpen as a prop then?

Reply:
Yes you can.